spark_sql/src/main/spark2.1/src/main/java/com/splicemachine/sparksql/SparkSQLUtilsImpl.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2012 - 2021 Splice Machine, Inc.
+ *
+ * This file is part of Splice Machine.
+ * Splice Machine is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3, or (at your option) any later version.
+ * Splice Machine is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License along with Splice Machine.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.splicemachine.sparksql;
+
+
+import com.splicemachine.utils.SparkSQLUtils;
+import org.apache.spark.SparkContext;
+
+/* Purpose:  Provide interfaces to access Spark objects or APIs
+ *           which may differ depending on the Spark version.
+ */
+public class SparkSQLUtilsImpl implements SparkSQLUtils {
```

Comment:
The implementation for all platforms is the same, it should be in `spark_sql/src/main/java` only once instead of copied in multiple platforms.

Reply:
Good point.  I've made the change.  If we need to have different method implementations in the future based on spark version, we could create separate source files for it at that time.